cdm/src/main/java/thredds/crawlabledataset/s3/CachingThreddsS3Client.java

```diff
@@ -98,6 +108,41 @@ public ObjectListing listObjects(S3URI s3uri) {
         return objectListing.orNull();
     }
 
+    @Override
+    public ThreddsS3Metadata getMetadata(S3URI s3uri) {
+        Optional<ThreddsS3Metadata> metadata = metadataCache.getIfPresent(s3uri);
+        if (metadata == null) {
```

Comment:
I hate all the Optional stuff, when null is an implicit value of any Java reference. Verbose and hard to understand.